requestor/server.py

```diff
@@ -88,24 +88,37 @@ async def close_service_manager():
     await app.service_manager.close()
 
 
-async def get_user_id():
-    data = await request.json
+def get_user_id():
     try:
-        return data['user_id']
+        auth = request.headers['Authorization']
     except KeyError:
-        raise UserDataMissing
+        raise UserDataMissing('Missing authorization header')
 
+    if auth.startswith("Bearer "):
+        token = auth[7:]
+    else:
+        raise UserDataMissing('Authorization header should start with "Bearer: "')
 
-@app.route('/getInstances', methods=['POST'])
+    if len(token) < 10:
+        #   This token is surely invalid, and we don't know exactly how the correct
+        #   tokens look like -> this check is the best we can do
+        raise UserDataMissing('Token is expected to be at least 10 characters long')
```

Comment:
Wait, the *token* here is ethereum address. So we know exactly what the length should be. It's 42 characters.